central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -638,19 +624,29 @@ func (s *flowStoreImpl) RemoveMatchingFlows(ctx context.Context, keyMatchFn func
 		}
 
 		for _, flow := range deleteFlows {
-			_, err := tx.Exec(ctx, deleteStmt, flow.GetProps().GetSrcEntity().GetType(), flow.GetProps().GetSrcEntity().GetId(), flow.GetProps().GetDstEntity().GetType(), flow.GetProps().GetDstEntity().GetId(), flow.GetProps().GetDstPort(), flow.GetProps().GetL4Protocol(), s.clusterID)
+			if valueMatchFn != nil {
+				if !valueMatchFn(flow) {
+					continue
+				}
+			}
+			// This is a cleanup operation so we can make it slow for now
+			tx, err := conn.Begin(ctx)
```

Comment:
Curious, why do we process it one by one? How long/big is one delete operation here?

Reply:
@c-du how long is hard to say.  The problem is that the scale tests exposed some issues where we could be processing inserts and deployment deletions at the same time.  Or we could be processing RemoveFlowsForDeployment at the same time as RemoveMatchingFlows.  Those exposed deadlock conditions as we would run into row lock condition.  The simplest solution, at least for me, in the short term was to just slow these down to one at a time as these are not time critical.  Then in ROX-9921 we can hopefully adjust how the entire flows work and resolve some of these type of issues.